src/ibex_bluesky_core/callbacks/__init__.py

```diff
@@ -1 +1,203 @@
+                ...
+
+                @icc
+                def _inner():
+                    yield from ...
+                    ...
+                    print(icc.live_fit.result.fit_report())
+                    print(f"COM: {icc.peak_stats['com']}")
+
+        Args:
+            x: The signal name to use for X within plots and fits.
+            y: The signal name to use for Y within plots and fits.
+            yerr: The signal name to use for the Y uncertainty within plots and fits.
+            fit: The fit method to use when fitting.
+            measured_fields: the fields to use for both the live table and human-readable file.
+            fields_for_live_table: the fields to measure for the live table (in addition to `measured_fields`).
+            fields_for_hr_file: the fields to measure for the human-readable file (in addition to `measured_fields`).
+            human_readable_file_output_dir: the output directory for human-readable files. can be blank and will default.
+            add_human_readable_file_cb: whether to add a human-readable file callback.
+            add_plot_cb: whether to add a plot callback.
+            add_fit_cb: whether to add a fitting callback (which will be displayed on a plot)
+            add_table_cb: whether to add a table callback.
+            add_peak_stats: whether to add a peak stats callback.
+            show_fit_on_plot: whether to show fit on plot.
+            ax: An optional axes object to use for plotting.
+
+        """  # noqa
+        if measured_fields is None:
+            measured_fields = []
+        if fields_for_live_table is None:
+            fields_for_live_table = []
+        if fields_for_hr_file is None:
+            fields_for_hr_file = []
+
+        if show_fit_on_plot and (not add_fit_cb or fit is None):
+            raise ValueError(
+                "Fit has been requested to show on plot without a fitting method or callback."
+            )
+        if (add_peak_stats or add_fit_cb or add_plot_cb) and (x is None or y is None):
+            raise ValueError(
+                "X and/or Y not specified when trying to add a plot, fit or peak stats."
+            )
+
+        self.subs = []
+        if add_human_readable_file_cb:
+            combined_hr_fields = measured_fields + fields_for_hr_file
+            if not combined_hr_fields:
+                raise ValueError("No fields specified for the human-readable file")
+            self.subs.append(
+                HumanReadableFileCallback(
+                    fields=combined_hr_fields,
+                    output_dir=Path(human_readable_file_output_dir)
+                    if human_readable_file_output_dir
+                    else DEFAULT_PATH,
+                ),
+            )
+
+        if add_table_cb:
+            combined_lt_fields = measured_fields + fields_for_live_table
+            if not combined_lt_fields:
+                raise ValueError("No fields specified for the live table")
```

Comment:
I think we mandate at least `x` and `y` for this callback collection, and then this can never occur